src/sys/unix.rs

```diff
@@ -526,11 +526,22 @@ pub fn preferred_languages(arena: &Arena) -> Vec<ArenaString<'_>, &Arena> {
     let mut locales = Vec::new_in(arena);
 
     for key in ["LANGUAGE", "LC_ALL", "LANG"] {
-        if let Ok(val) = std::env::var(key) {
-            locales.extend(
-                val.split(':').filter(|s| !s.is_empty()).map(|s| ArenaString::from_str(arena, s)),
-            );
-            break;
+        if let Ok(langs) = std::env::var(key) {
+            let langs = langs
+                // split colon separated entries
+                .split(':')
+                // strip territory ("EN" in "en_EN")
+                .map(|s| s.split('_').next().unwrap_or(s))
+                // strip codeset ("utf8" in "C.utf8")
+                .map(|s| s.split('.').next().unwrap_or(s))
```

Comment:
This isn't the right place to do this. The reason matching doesn't work right now is because of #74. The issue shows where the fix needs to be made (= needs prefix match).

The only change needed in this function IMO is swapping the unconditional `break` with the `if` condition you added below.

Reply:
I just came across #85 which actually almost fixes this bug (only the case-insensitivity is missing). As such, I think this PR only needs to fix the early-exit?

Reply:
> I just came across #85 which actually almost fixes this bug (only the case-insensitivity is missing). As such, I think this PR only needs to fix the early-exit?

Done

Reply:
Hmm that's odd. I'll try to test both of your PRs locally as soon as I can.